pkgs/by-name/fi/fido2-manage/package.nix

```diff
@@ -0,0 +1,138 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  pkg-config,
+  cmake,
+  libcbor,
+  openssl,
+  zlib,
+  gnugrep,
+  gawk,
+  # Linux only
+  pcsclite,
+  udev,
+  # GUI
+  python3,
+  xterm,
+  makeDesktopItem,
+  copyDesktopItems,
+  # Darwin only
+  libuv,
+  libsolv,
+  libcouchbase,
+  darwin,
+}:
+let
+  pythonEnv = python3.withPackages (ps: [ ps.tkinter ]);
+in
+stdenv.mkDerivation rec {
+  pname = "fido2-manage";
+  version = "0-unstable-2024-09-24";
+
+  src = fetchFromGitHub {
+    owner = "token2";
+    repo = "fido2-manage";
+    rev = "6aef9ceccdf7bcc60a9298e51a4db633256925df";
+    hash = "sha256-rXTL6wpdvCifakmxH14wBLbhTptNYNFGEPskpUy3IjA=";
+  };
+
+  nativeBuildInputs =
+    [
+      pkg-config
+      cmake
+    ]
+    ++ lib.optionals stdenv.isLinux [
+      copyDesktopItems
+    ];
+
+  buildInputs =
+    [
+      libcbor
+      openssl
+      zlib
+    ]
+    ++ lib.optionals stdenv.isLinux [
+      xterm
+      udev
+      pcsclite
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      libuv
+      libsolv
+      libcouchbase
+      gnugrep
+      darwin.apple_sdk.frameworks.IOKit
+      darwin.apple_sdk.frameworks.PCSC
+    ];
+
+  cmakeFlags = [ "-USE_PCSC=ON" ];
+
+  postPatch =
+    ''
+      substituteInPlace ./src/libfido2.pc.in \
+        --replace-fail "\''${prefix}/@CMAKE_INSTALL_LIBDIR@" "@CMAKE_INSTALL_FULL_LIBDIR@"
+    ''
+    + lib.optionalString stdenv.isDarwin ''
+      substituteInPlace ./CMakeLists.txt \
+        --replace-fail "/\''${CMAKE_INSTALL_LIBDIR}" "/lib"
+    '';
+
+  postInstall =
+    lib.optionalString stdenv.isLinux ''
+      install $src/fido2-manage.sh $out/bin/${pname}
```

Comment:
Please don't substitute pname for the binary path, just hardcode the string instead

Reply:
I removed all pname references. Just for my understanding: what is the reasoning behind this? Evaluation time?